src/Features/CSharp/Portable/Diagnostics/Analyzers/CSharpPreferFrameworkTypeDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Diagnostics.PreferFrameworkType;
+
+namespace Microsoft.CodeAnalysis.CSharp.Diagnostics.Analyzers
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    internal class CSharpPreferFrameworkTypeDiagnosticAnalyzer : PreferFrameworkTypeDiagnosticAnalyzerBase<SyntaxKind>
+    {
+        protected override ImmutableArray<SyntaxKind> SyntaxKindsOfInterest => 
+            ImmutableArray.Create(SyntaxKind.PredefinedType);
+
+        protected override bool IsPredefinedTypeAndReplaceableWithFrameworkType(SyntaxNode node)
+        {
+            var syntaxKind = (node as PredefinedTypeSyntax)?.Keyword.Kind();
+
+            // every predefined type keyword except `void` can be replaced by its framework type in code.
+            return syntaxKind != null && 
+                   syntaxKind != SyntaxKind.VoidKeyword && 
+                   SyntaxFacts.IsPredefinedType(syntaxKind.Value);
+        }
+
+        protected override bool IsInMemberAccessContext(SyntaxNode node, SemanticModel semanticModel)
+        {
+            var expression = (node as ExpressionSyntax);
+            if (expression == null)
+            {
+                return false;
+            }
+
+            return expression.IsInMemberAccessContext() || expression.InsideCrefReference();
```

Comment:
to fill in the context for other reviewers: The existing simplify to predefined type analyzer ties crefs to "In MemberAccess" option and not the "Declaration" option. By looking at the history of the source, I found it was done as a part of some bug fix. Now, since it works that way for converting to predefined types, I had to match the same behavior for converting to framework type analyzer.. Its a questionable behavior, but pre-existing and introduced new by this PR. I'll dig into the bug and provide more context as to why it is so.